### ISSUE TITLE
"console": "terminal" does not work with Flutter

### ISSUE THREAD

**Issue by IgorKrupenja** [Dart-Code/Dart-Code#3339]
**Describe the bug**
"console": "terminal" does nothing for a Flutter app, the output is still shown in Debug Console.

**To Reproduce**
1. Create a flutter app.
2. Open `launch.json` in VSCode
3. Add e.g. the following configuration:

```json
{
      "name": "Debug",
      "request": "launch",
      "type": "dart",
      "args": ["--verbose"],
      "console": "terminal"
},
```

4. Run the app with this configuration either from Debug panel or by selecting "Debug: select and start debugging" in command palette.

**Expected behavior**
The output should be shown in Terminal. Instead, it is still shown in Debug console. Interestingly, other configurations (like `"args": ["--verbose"]`) seem to work fine.

**Screenshots**
![image](https://user-images.githubusercontent.com/26354715/118137669-24826800-b40e-11eb-9482-3c6624e89e6e.png)

**Versions (please complete the following information):**
 - VS Code version: 1.56.1
 - Dart extension version: 3.22.0
 - Dart/Flutter SDK version: Dart 2.12.3, Flutter 2.0.6

**Comment by DanTup** [comment]
Can you confirm the reason you'd like to run in the terminal for Flutter? This isn't currently possible and I'm not sure that's likely to change, as the way Flutter apps runs is quite different, for ex.:

- The editor communicates with Flutter over stdin/stdout, so the debug adapter needs to have access to these streams (when running in the terminal, the debug adapter has no access to the process streams)
- For Flutter, your Dart code may be running on a remote device so the stdin stream of the Flutter process is not for the VM running your code (so if you're trying to accept input, this might not work unless Flutter was proxying input to the process)

Solving these issues would involve changes to both the VS Code extension and Flutter tooling so there would likely need to be a compelling reason to handle this. Perhaps there's another way to achieve what you need though?

**Comment by igor-krupenja** [comment]
The main reason I want to run in terminal is that VSCode Terminal supports search and Debug Console does not (it has filtering but filtering does not give context). In this case, I guess I will simply stick to Android Studio which has search in console.

**Comment by DanTup** [comment]
Thanks for the clarification. Although I agree the lack of search is frustrating, it seems like having a search in the VS Code Debug Console would be a better fix than trying to handle this another way (and solve the complications mentioned above). There is an open issue about this in VS Code (https://github.com/microsoft/vscode/issues/4338) worth 👍 ing, although since it was added to the "On Deck" milestone over a year ago, I'm not sure if it's in the near-term plans.

I'll leave this issue open in the backlog for now to collect 👍 s. If there's a lot of demand and VS Code doesn't resolve this, we can consider coming up with another solution.

**Comment by igor-krupenja** [comment]
Yep, I actually agree with you that search in VSCode Console would be a better solution. Thank you for pointing me to the issue in VSCode repo, I have seen it before and already given the thumbs up there.

**Comment by DanTup** [comment]
Closing this since it would be very difficult to do and there wasn't a lot of interest here.
